pkg/maps/configmap/config_map.go

```diff
@@ -0,0 +1,187 @@
+}
+
+// String pretty print the ConfigKey
+func (k *ConfigKey) String() string {
+	return fmt.Sprintf("%d", k.key)
+}
+
+// String pretty print the ConfigValues.
+func (v *ConfigValues) String() string {
+	return v.value0.String()
+}
+
+// String pretty print the ConfigValue.
+func (v *ConfigValue) String() string {
+	return fmt.Sprintf("<NOT TO BE USED DIRECTLY>")
+}
+
+// String pretty print the ConfigValue0.
+func (v *ConfigValue0) String() string {
+	return fmt.Sprintf("time offset: %s", v.uTimeOffset.String())
+}
+
+// GetValuePtr returns the unsafe pointer to the BPF value.
+func (v *ConfigValue) GetValuePtr() unsafe.Pointer { return unsafe.Pointer(v) }
+
+// GetValuePtr returns the unsafe pointer to the BPF value.
+func (v *ConfigValue0) GetValuePtr() unsafe.Pointer { return unsafe.Pointer(v) }
+
+// GetKeyPtr returns the unsafe pointer to the BPF key
+func (k *ConfigKey) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+
+// NewValue returns a new empty instance of the structure represeting the BPF
+// map value
+func (k *ConfigKey) NewValue() bpf.MapValue { return &ConfigValue{} }
+
+func newConfigKey(key uint32) *ConfigKey {
+	return &ConfigKey{
+		key: key,
+	}
+}
+
+type ConfigMap struct {
+	*bpf.Map
+	config ConfigValues // current config
+}
+
+var (
+	once      sync.Once
+	configMap ConfigMap
+)
+
+type ControllerManager interface {
+	UpdateController(name string, params controller.ControllerParams)
+}
+
+// MapCreate will create an config map
+func InitMap(ctx context.Context, controllerManager ControllerManager) error {
+	once.Do(func() {
+		configMap = ConfigMap{
+			Map: bpf.NewMap(MapName,
+				bpf.MapTypeArray,
+				&ConfigKey{},
+				int(unsafe.Sizeof(ConfigKey{})),
+				&ConfigValue{},
+				int(unsafe.Sizeof(ConfigValue{})),
+				MaxEntries,
+				0, 0,
+				bpf.ConvertKeyValue,
+			),
+		}
+
+		configMap.syncConfig(true)
+
+		// Add controller for keeping clock in sync for NTP time jumps and any difference
+		// between monotonic and boottime clocks.
+		ctrlName := "sync-config"
+		controllerManager.UpdateController(ctrlName,
+			controller.ControllerParams{
+				DoFunc: func(ctx context.Context) error {
+					return configMap.syncConfig(false)
+				},
+				RunInterval: 1 * time.Minute,
+				Context:     ctx,
+			},
+		)
+	})
+
+	_, err := configMap.OpenOrCreate()
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+// updateConfig writes the current config to the bpf map.
+func (r *ConfigMap) updateConfig(config *ConfigValues, force bool) (err error) {
+	// Currently only handles the first 64 bytes of the config. Adjust when more is needed.
+	value := config.value0
+	k := newConfigKey(0)
+	if force || value != r.config.value0 {
+		err = r.Update(k, &value)
+		if err == nil {
+			// Update cached value on success
+			r.config.value0 = value
+		}
+	}
+	return err
+}
+
+// syncConfig updates the current config if needed
+func (r *ConfigMap) syncConfig(force bool) error {
+	_, err := r.OpenOrCreate()
```

Comment:
Why are we calling `OpenOrCreate` again here? It was already called from `InitMap`